crates/autopilot/src/database/auction.rs

```diff
@@ -0,0 +1,161 @@
+use super::Postgres;
+use anyhow::{anyhow, Context, Result};
+use database::{
+    auction::AuctionId,
+    orders::{
+        BuyTokenDestination as DbBuyTokenDestination, OrderKind as DbOrderKind,
+        SellTokenSource as DbSellTokenSource, SigningScheme as DbSigningScheme,
+    },
+    solver_competition::SolverCompetitionId,
+};
+use futures::{StreamExt, TryStreamExt};
+use model::{
+    app_id::AppId,
+    auction::Auction,
+    order::{
+        BuyTokenDestination, Order, OrderData, OrderKind, OrderMetadata, OrderStatus, OrderUid,
+        SellTokenSource,
+    },
+    signature::{Signature, SigningScheme},
+};
+use number_conversions::{big_decimal_to_big_uint, big_decimal_to_u256};
+use primitive_types::H160;
+
+pub struct SolvableOrders {
+    pub orders: Vec<Order>,
+    pub latest_settlement_block: u64,
+}
+
+impl Postgres {
+    pub async fn solvable_orders(&self, min_valid_to: u32) -> Result<SolvableOrders> {
+        let _timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["solvable_orders"])
+            .start_timer();
+
+        let mut ex = self.0.begin().await?;
+        let orders = database::orders::solvable_orders(&mut ex, min_valid_to as i64)
+            .map(|result| match result {
+                Ok(order) => full_order_into_model_order(order),
+                Err(err) => Err(anyhow::Error::from(err)),
+            })
+            .try_collect::<Vec<_>>()
+            .await?;
+        let latest_settlement_block =
+            database::orders::latest_settlement_block(&mut ex).await? as u64;
+        Ok(SolvableOrders {
+            orders,
+            latest_settlement_block,
+        })
+    }
+
+    pub async fn replace_current_auction(&self, auction: &Auction) -> Result<AuctionId> {
+        let _timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["save_auction"])
+            .start_timer();
+
+        let data = serde_json::to_value(&auction)?;
+        let mut ex = self.0.begin().await?;
+        database::auction::delete_all_auctions(&mut ex).await?;
```

Comment:
It's okay to delete all the auctions here because they will end up in the settlement competition table eventually, right?

Reply:
I also found the fact that we can delete all auctions surprising.

I know that we store he settlement competition information in the database, and it includes order UIDs that were included in the auction - but maybe it does make sense to also include this additional data (because things like available balance, executed amounts, etc. get lost).

Reply:
At the moment this table just facilitates communicating the auction from the autopilot to the orderbook and giving a persistently increasing id to the autopilot. The settlement competition table still exists the way it works isn't changed in this PR. That means there is no reason to persist these auctions.

Eventually I could see us combining the tables in some way but I'm not sure about the details yet.